anycloud/cli/src/main.rs

```diff
@@ -117,6 +134,57 @@ pub async fn main() {
       }
     }
     // rely on AppSettings::SubcommandRequiredElseHelp
-    _ => {}
+    _ => {
+      authenticate(false).await;
+    }
   }
 }
+
+async fn new_or_upgrade(
+  mode: &str,
+  anycloud_agz: String,
+  env_file: Option<&str>,
+  app_name: Option<&str>,
+  config_name: Option<&str>,
+  non_interactive: bool,
+) {
+  let dockerfile_b64 = get_dockerfile_b64().await;
+  let app_tar_gz_b64 = get_app_tar_gz_b64(true).await;
+  let env_b64 = match env_file {
+    Some(env_file) => Some(get_env_file_b64(env_file.to_string()).await),
+    None => None,
+  };
+  let app_name = match app_name {
+    Some(name) => Some(name.to_string()),
+    None => None,
+  };
+  let config_name = match config_name {
+    Some(name) => Some(name.to_string()),
+    None => None,
+  };
+  match mode {
+    "new" => {
+      deploy::new(
+        anycloud_agz,
+        Some((dockerfile_b64, app_tar_gz_b64)),
+        env_b64,
+        app_name,
+        config_name,
+        non_interactive,
+      )
+      .await
+    }
+    "upgrade" => {
+      deploy::upgrade(
+        anycloud_agz,
+        Some((dockerfile_b64, app_tar_gz_b64)),
+        env_b64,
+        app_name,
+        config_name,
+        non_interactive,
+      )
+      .await
+    }
+    _ => {}
+  };
```

Comment:
Nit: Make this an enum instead of a string and you don't need the `_ => {}` bit at the end?

Alternatively, instead of `mode: &str` you pass the async function to call and you don't need the match at all? (That one is more clever but less extensible in the future if the arguments ever change between one or the other.)

Reply:
I'll go with the latest alternative. Do not see this two being too different in the future